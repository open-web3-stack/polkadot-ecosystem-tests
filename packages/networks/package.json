{
	"name": "@e2e-test/networks",
	"type": "module",
	"dependencies": {
		"@acala-network/chopsticks-testing": "^0.13.1",
		"@e2e-test/shared": "workspace:*"
	},
	"main": "./dist/cjs/index.js",
	"module": "./dist/esm/index.js",
	"files": [
		"*",
		"!tsconfig.json",
		"!**/*.test.ts"
	],
	"exports": {
		".": {
			"require": "./dist/cjs/index.js",
			"import": "./dist/esm/index.js",
			"default": "./dist/esm/index.js"
		},
		"./chains": {
			"require": "./dist/cjs/chains/index.js",
			"import": "./dist/esm/chains/index.js",
			"default": "./dist/esm/chains/index.js"
		},
		"./*": {
			"require": "./dist/cjs/*.js",
			"import": "./dist/esm/*.js",
			"default": "./dist/esm/*.js"
		}
	}
}
