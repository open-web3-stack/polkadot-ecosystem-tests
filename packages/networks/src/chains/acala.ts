import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  acala: {
    relayToken: 'DOT',
    relayLiquidToken: 'LDOT',
    stableToken: 'AUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    dot: { Token: 'DOT' },
    ldot: { Token: 'LDOT' },
    dai: { Erc20: '0x54a37a01cd75b616d63e0ab665bffdb0143c52ae' },
    wbtc: { ForeignAsset: 5 },
    ausd: { Token: 'AUSD' },
    aca: { Token: 'ACA' },
    lcdot: { LiquidCrowdloan: 13 },
  },
  karura: {
    relayToken: 'KSM',
    relayLiquidToken: 'LKSM',
    stableToken: 'KUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    ksm: { Token: 'KSM' },
    lksm: { Token: 'LKSM' },
    usdt: { ForeignAsset: 7 },
    rmrk: { ForeignAsset: 0 },
    dai: { Erc20: '0x4bb6afb5fa2b07a5d1c499e1c3ddb5a15e709a71' },
    ausd: { Token: 'KUSD' },
    kar: { Token: 'KAR' },
  },
}

const getInitStorages = (config: typeof custom.acala | typeof custom.karura) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 4, data: { free: 10 * 1e12 } }]],
  },
  Tokens: {
    accounts: [
      [[defaultAccount.alice.address, { Token: config.relayToken }], { free: 10 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.relayLiquidToken }], { free: 100 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.stableToken }], { free: 1000 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccount.alice.address,
  },
  EvmAccounts: {
    accounts: [[['0x82a258cb20e2adb4788153cd5eb5839615ece9a0'], defaultAccount.alice.address]],
    evmAddresses: [[[defaultAccount.alice.address], '0x82a258cb20e2adb4788153cd5eb5839615ece9a0']],
  },
  Homa: {
    // avoid impact test outcome
    $removePrefix: ['redeemRequests', 'unbondings', 'toBondPool'],
    // so that bump era won't trigger unbond
    relayChainCurrentEra: 100,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const acala = defineChain({
  name: 'acala',
  endpoint: 'wss://acala-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.acala,
  initStorages: getInitStorages(custom.acala),
})

export const karura = defineChain({
  name: 'karura',
  endpoint: 'wss://karura-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.karura,
  initStorages: getInitStorages(custom.karura),
})
