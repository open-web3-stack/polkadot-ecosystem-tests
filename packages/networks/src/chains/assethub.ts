import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
  assetHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubPolkadot | typeof custom.assetHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubPolkadot = defineChain({
  name: 'assetHubPolkadot',
  endpoint: 'wss://rpc-asset-hub-polkadot.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubPolkadot,
  initStorages: getInitStorages(custom.assetHubPolkadot),
})

export const assetHubKusama = defineChain({
  name: 'assetHubKusama',
  endpoint: 'wss://rpc-asset-hub-kusama.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubKusama,
  initStorages: getInitStorages(custom.assetHubKusama),
})
