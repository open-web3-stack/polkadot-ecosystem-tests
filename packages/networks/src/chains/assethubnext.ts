import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNextWestend: {
    wnd: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNextWestend) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNextWestend = defineChain({
  name: 'assetHubNextWestend',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNextWestend,
  initStorages: getInitStorages(custom.assetHubNextWestend),
})
