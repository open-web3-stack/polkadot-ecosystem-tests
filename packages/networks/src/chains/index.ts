export * from './acala.js'
export * from './assethub.js'
export * from './astar.js'
export * from './hydraDX.js'
export * from './moonbeam.js'
export * from './polkadot.js'
