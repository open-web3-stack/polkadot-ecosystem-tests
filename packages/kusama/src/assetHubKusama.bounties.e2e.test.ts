import { assetHubKusama } from '@e2e-test/networks/chains'
import { baseBountiesE2ETests, type ParaTestConfig, registerTestTree } from '@e2e-test/shared'

const testConfig: ParaTestConfig = {
  testSuiteName: 'Kusama Asset Hub Bounties',
  addressEncoding: 2,
  blockProvider: 'NonLocal',
  asyncBacking: 'Enabled',
}

registerTestTree(baseBountiesE2ETests(assetHubKusama, testConfig))
