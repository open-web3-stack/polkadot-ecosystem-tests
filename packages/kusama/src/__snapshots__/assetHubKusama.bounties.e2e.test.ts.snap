// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Kusama Asset Hub Bounties > All bounty failure tests > Bounty cannot be awarded if it has an active child bounty > child bounty added events 1`] = `
[
  {
    "data": "(redacted)",
    "method": "Added",
    "section": "childBounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty failure tests > Bounty cannot be claimed in active state > bounty approved events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyApproved",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty failure tests > Bounty cannot be claimed in active state > bounty became active events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyBecameActive",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty failure tests > Bounty cannot be claimed in active state > bounty proposed events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty failure tests > Bounty cannot be claimed in active state > curator accepted events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorAccepted",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty failure tests > Bounty cannot be claimed in active state > curator proposed events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty failure tests > Bounty closure in approved state > scheduler events when closing bounty with approved state fails 1`] = `
[
  {
    "data": {
      "id": null,
      "result": {
        "Err": {
          "Module": {
            "error": "0x03000000",
            "index": 95,
          },
        },
      },
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty failure tests > Bounty closure in pending payout state > scheduler events when closing bounty with pending payout fails 1`] = `
[
  {
    "data": {
      "id": null,
      "result": {
        "Err": {
          "Module": {
            "error": "0x07000000",
            "index": 95,
          },
        },
      },
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in Active state by Treasurer > Bounty bond is deposited to the treasury 1`] = `
[
  {
    "data": "(redacted)",
    "method": "Deposit",
    "section": "treasury",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in Active state by Treasurer > bounty approved events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyApproved",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in Active state by Treasurer > bounty became active events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyBecameActive",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in Active state by Treasurer > bounty proposed events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in Active state by Treasurer > curator accepted events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorAccepted",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in Active state by Treasurer > curator proposed events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in Active state by Treasurer > curator slash event 1`] = `
[
  {
    "data": {
      "amount": "(rounded 170000000000)",
      "who": "(redacted)",
    },
    "method": "Slashed",
    "section": "balances",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in Active state by Treasurer > curator unassigned active by treasurer events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
    },
    "method": "CuratorUnassigned",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in Active state by curator themselves > bounty approved events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyApproved",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in Active state by curator themselves > bounty became active events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyBecameActive",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in Active state by curator themselves > bounty proposed events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in Active state by curator themselves > curator accepted events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorAccepted",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in Active state by curator themselves > curator proposed events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in Active state by curator themselves > curator unassigned active by curator events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
    },
    "method": "CuratorUnassigned",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in ApprovedWithCurator state > curator unassigned approved with curator events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
    },
    "method": "CuratorUnassigned",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in CuratorProposed state > bounty approved events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyApproved",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in CuratorProposed state > bounty became active events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyBecameActive",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in CuratorProposed state > curator unassigned curator proposed events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
    },
    "method": "CuratorUnassigned",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in PendingPayout state > Bounty bond is deposited to the treasury 1`] = `
[
  {
    "data": "(redacted)",
    "method": "Deposit",
    "section": "treasury",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in PendingPayout state > bounty approved events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyApproved",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in PendingPayout state > bounty awarded events 1`] = `
[
  {
    "data": {
      "beneficiary": "D8ew585BL5H1ALhn4kmJoxhgqcgKDZLPc6xunJNv4mmrBns",
      "index": "(redacted)",
    },
    "method": "BountyAwarded",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in PendingPayout state > bounty became active events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyBecameActive",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in PendingPayout state > bounty proposed events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in PendingPayout state > curator accepted events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorAccepted",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in PendingPayout state > curator proposed events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in PendingPayout state > curator slash event 1`] = `
[
  {
    "data": {
      "amount": "(rounded 170000000000)",
      "who": "(redacted)",
    },
    "method": "Slashed",
    "section": "balances",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > All curator unassign tests > Unassign curator in PendingPayout state > curator unassigned pending payout events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
    },
    "method": "CuratorUnassigned",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty Closure Tests > Bounty closure in active state > Bounty value is transferred to the treasury 1`] = `
[
  {
    "data": {
      "amount": "(rounded 67000000000000)",
      "from": "(redacted)",
      "to": "(redacted)",
    },
    "method": "Transfer",
    "section": "balances",
  },
  {
    "data": {
      "amount": "(rounded 44000000)",
      "from": "(redacted)",
      "to": "(redacted)",
    },
    "method": "Transfer",
    "section": "balances",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty Closure Tests > Bounty closure in active state > bounty approved events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyApproved",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty Closure Tests > Bounty closure in active state > bounty became active events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyBecameActive",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty Closure Tests > Bounty closure in active state > bounty canceled event 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyCanceled",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty Closure Tests > Bounty closure in active state > curator accepted events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorAccepted",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty Closure Tests > Bounty closure in active state > curator proposed events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty Closure Tests > Bounty closure in funded state > bounty approved events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyApproved",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty Closure Tests > Bounty closure in funded state > bounty became active events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyBecameActive",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty Closure Tests > Bounty closure in funded state > bounty canceled events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyCanceled",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty Closure Tests > Bounty closure in funded state > bounty value transferred to treasury 1`] = `
[
  {
    "data": {
      "amount": "(rounded 67000000000000)",
      "from": "(redacted)",
      "to": "(redacted)",
    },
    "method": "Transfer",
    "section": "balances",
  },
  {
    "data": {
      "amount": "(rounded 6300000)",
      "from": "(redacted)",
      "to": "(redacted)",
    },
    "method": "Transfer",
    "section": "balances",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty Closure Tests > Bounty closure in proposed state > bounty rejected events 1`] = `
[
  {
    "data": {
      "bond": "(rounded 72000000)",
      "index": "(redacted)",
    },
    "method": "BountyRejected",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty Closure Tests > Bounty closure in proposed state > proposer bond slashed event 1`] = `
[
  {
    "data": {
      "amount": "(rounded 72000000)",
      "who": "(redacted)",
    },
    "method": "Slashed",
    "section": "balances",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty approval tests > Bounty approval flow > bounty approval events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyApproved",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty approval tests > Bounty approval flow with curator > bounty approval with curator events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyApproved",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty approval tests > Bounty approval flow with curator > curator proposed events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty awarding and claiming > bounty approved events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyApproved",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty awarding and claiming > bounty awarded events 1`] = `
[
  {
    "data": {
      "beneficiary": "D8ew585BL5H1ALhn4kmJoxhgqcgKDZLPc6xunJNv4mmrBns",
      "index": "(redacted)",
    },
    "method": "BountyAwarded",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty awarding and claiming > bounty became active events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyBecameActive",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty awarding and claiming > bounty claimed events 1`] = `
[
  {
    "data": {
      "beneficiary": "D8ew585BL5H1ALhn4kmJoxhgqcgKDZLPc6xunJNv4mmrBns",
      "index": "(redacted)",
      "payout": "(rounded 60000000000000)",
    },
    "method": "BountyClaimed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty awarding and claiming > bounty proposed events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty awarding and claiming > curator accepted events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorAccepted",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty awarding and claiming > curator proposed events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty extension > bounty approved events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyApproved",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty extension > bounty became active events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyBecameActive",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty extension > bounty extended events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyExtended",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty extension > bounty proposed events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty extension > curator accepted events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorAccepted",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty extension > curator proposed events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty funding tests > Bounty funding for Approved Bounties > bounty approved events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyApproved",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty funding tests > Bounty funding for Approved Bounties > bounty became active events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyBecameActive",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty funding tests > Bounty funding for Approved Bounties > bounty proposed events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty funding tests > Bounty funding for ApprovedWithCurator Bounties > bounty approved events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyApproved",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty funding tests > Bounty funding for ApprovedWithCurator Bounties > bounty became active events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyBecameActive",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty funding tests > Bounty funding for ApprovedWithCurator Bounties > bounty proposed events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Bounty funding tests > Bounty funding for ApprovedWithCurator Bounties > curator proposed events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Creating a bounty > bounty proposal events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Curator assignment and acceptance > bounty approved events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyApproved",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Curator assignment and acceptance > bounty became active events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyBecameActive",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Curator assignment and acceptance > bounty proposed events 1`] = `
[
  {
    "data": {
      "index": "(redacted)",
    },
    "method": "BountyProposed",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Curator assignment and acceptance > curator accepted events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorAccepted",
    "section": "bounties",
  },
]
`;

exports[`Kusama Asset Hub Bounties > All bounty success tests > Curator assignment and acceptance > curator proposed events 1`] = `
[
  {
    "data": {
      "bountyId": "(redacted)",
      "curator": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "CuratorProposed",
    "section": "bounties",
  },
]
`;
